server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1026,10 +1044,13 @@ private boolean checkForGuestsUpdate(Consumer existing, Consumer incoming) {
             }
 
             // The guest has not registered. No need to process entitlements.
-            if (guest == null) {
+            if (guestConsumerMap.get(guestId.getGuestId()) == null) {
                 continue;
             }
 
+            Consumer guest = guestConsumerMap.get(guestId.getGuestId());
```

Comment:
Could avoid a 2nd map lookup by moving this above the if -- just being picky cuz I can't find anything else. :)